src/CommunityToolkit.Aspire.Hosting.OpenTelemetryCollector/CollectorExtensions.cs

```diff
@@ -0,0 +1,105 @@
+        string name,
+        Action<OpenTelemetryCollectorSettings>? configureSettings = null)
+    {
+        var url = builder.Configuration[DashboardOtlpUrlVariableName] ??
+            builder.Configuration[DashboardOtlpUrlVariableNameLegacy] ??
+            DashboardOtlpUrlDefaultValue;
+
+        var settings = new OpenTelemetryCollectorSettings();
+        configureSettings?.Invoke(settings);
+
+        var isHttpsEnabled = !settings.ForceNonSecureReceiver && url.StartsWith("https", StringComparison.OrdinalIgnoreCase);
+
+        var dashboardOtlpEndpoint = ReplaceLocalhostWithContainerHost(url, builder.Configuration);
+
+        var resource = new CollectorResource(name);
+        var resourceBuilder = builder.AddResource(resource)
+            .WithImage(settings.CollectorImage, settings.CollectorVersion)
+            .WithEnvironment("ASPIRE_ENDPOINT", dashboardOtlpEndpoint)
+            .WithEnvironment("ASPIRE_API_KEY", builder.Configuration[DashboardOtlpApiKeyVariableName]);
+
+        if (settings.EnableGrpcEndpoint)
+            resourceBuilder.WithEndpoint(targetPort: 4317, name: CollectorResource.GRPCEndpointName, scheme: isHttpsEnabled ? "https" : "http");
+        if (settings.EnableHttpEndpoint)
+            resourceBuilder.WithEndpoint(targetPort: 4318, name: CollectorResource.HTTPEndpointName, scheme: isHttpsEnabled ? "https" : "http");
+
+
+        if (!settings.ForceNonSecureReceiver && isHttpsEnabled && builder.ExecutionContext.IsRunMode && builder.Environment.IsDevelopment())
+        {
+            DevCertHostingExtensions.RunWithHttpsDevCertificate(resourceBuilder, "HTTPS_CERT_FILE", "HTTPS_CERT_KEY_FILE", (certFilePath, certKeyPath) =>
+            {
+                if (settings.EnableHttpEndpoint)
+                {
+                    resourceBuilder.WithArgs(
+                        $@"--config=yaml:receivers::otlp::protocols::http::tls::cert_file: ""{certFilePath}""",
+                        $@"--config=yaml:receivers::otlp::protocols::http::tls::key_file: ""{certKeyPath}""");
+                }
+                if (settings.EnableGrpcEndpoint)
+                {
+                    resourceBuilder.WithArgs(
+                        $@"--config=yaml:receivers::otlp::protocols::grpc::tls::cert_file: ""{certFilePath}""",
+                        $@"--config=yaml:receivers::otlp::protocols::grpc::tls::key_file: ""{certKeyPath}""");
+                }
+            });
+        }
+        return resourceBuilder;
+    }
+
+    /// <summary>
+    /// Force all apps to forward to the collector instead of the dashboard directly
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    public static IResourceBuilder<CollectorResource> WithAppForwarding(this IResourceBuilder<CollectorResource> builder)
+    {
+        builder.ApplicationBuilder.Services.TryAddLifecycleHook<EnvironmentVariableHook>();
```

Comment:
Lifecycle hooks are mostly considered legacy with the event system now in place. The hook should be replacable with the `OnAfterEndpointsAllocated` callback.